sdk/core/core/src/az_client_secret_credential.c

```diff
@@ -43,125 +43,75 @@ static AZ_NODISCARD az_result az_token_credential_get_token(
-
-    size_t const request_elements[] = {
-      credential->tenant_id.size * AZ_TOKEN_CREDENTIAL_GET_TOKEN_URLENCODE_FACTOR,
-      credential->client_id.size * AZ_TOKEN_CREDENTIAL_GET_TOKEN_URLENCODE_FACTOR,
-      credential->client_secret.size * AZ_TOKEN_CREDENTIAL_GET_TOKEN_URLENCODE_FACTOR,
-      resource_url.size * AZ_TOKEN_CREDENTIAL_GET_TOKEN_URLENCODE_FACTOR,
-    };
-
-    size_t required_request_size = auth_url1.size + auth_url2.size + auth_body1.size
-        + auth_body2.size + auth_body3.size + headers_size;
-
-    for (size_t i = 0; i < AZ_ARRAY_SIZE(request_elements); ++i) {
-      required_request_size += request_elements[i];
-      AZ_CONTRACT(required_request_size > request_elements[i], AZ_ERROR_BUFFER_OVERFLOW);
+      auth_url = az_span_builder_result(&builder);
     }
 
-    AZ_CONTRACT(response_buf.size >= required_request_size, AZ_ERROR_BUFFER_OVERFLOW);
-  }
+    uint8_t auth_body_buf[200] = { 0 };
+    az_span auth_body = { 0 };
 
-  {
-    AZ_CONTRACT(!az_span_is_overlap(az_mut_span_to_span(response_buf), resource_url), AZ_ERROR_ARG);
+    static az_span const auth_body1 = AZ_CONST_STR("grant_type=client_credentials&client_id=");
+    static az_span const auth_body2 = AZ_CONST_STR("&client_secret=");
+    static az_span const auth_body3 = AZ_CONST_STR("&resource=");
 
-    AZ_CONTRACT(
-        !az_span_is_overlap(az_mut_span_to_span(response_buf), credential->tenant_id),
-        AZ_ERROR_ARG);
-
-    AZ_CONTRACT(
-        !az_span_is_overlap(az_mut_span_to_span(response_buf), credential->client_id),
-        AZ_ERROR_ARG);
-
-    AZ_CONTRACT(
-        !az_span_is_overlap(az_mut_span_to_span(response_buf), credential->client_secret),
-        AZ_ERROR_ARG);
-  }
-
-  az_span auth_url = { 0 };
-  az_span auth_body = { 0 };
-  {
-    az_span_builder builder = az_span_builder_create(response_buf);
-
-    AZ_RETURN_IF_FAILED(az_span_builder_append(&builder, auth_url1));
-    AZ_RETURN_IF_FAILED(az_uri_encode(credential->tenant_id, &builder));
-    AZ_RETURN_IF_FAILED(az_span_builder_append(&builder, auth_url2));
-
-    auth_url = az_span_builder_result(&builder);
-    builder = az_span_builder_create(az_mut_span_drop(response_buf, auth_url.size + headers_size));
+    az_span_builder builder
+        = az_span_builder_create((az_mut_span)AZ_SPAN_FROM_ARRAY(auth_body_buf));
 
     AZ_RETURN_IF_FAILED(az_span_builder_append(&builder, auth_body1));
     AZ_RETURN_IF_FAILED(az_uri_encode(credential->client_id, &builder));
     AZ_RETURN_IF_FAILED(az_span_builder_append(&builder, auth_body2));
-    AZ_RETURN_IF_FAILED(az_uri_encode(credential->client_secret, &builder));
-    AZ_RETURN_IF_FAILED(az_span_builder_append(&builder, auth_body3));
-    AZ_RETURN_IF_FAILED(az_uri_encode(resource_url, &builder));
 
-    auth_body = az_span_builder_result(&builder);
-  }
+    az_mut_span secret_copy = { .begin = builder.buffer.begin + builder.size, .size = 0 };
```

Comment:
```c
size_t secret_start = builder.size;
AZ_RETURN_IF_FAILED(az_uri_encoding(credential->client_secret, &builder));
az_mut_span const secret_copy = az_mut_sub_span(builder.span, secret_start, builder.size);
```

Reply:
Done - the code is rewritten in a way it is no longer needed.